examples/autograd/doc_dist_train.py

```diff
@@ -0,0 +1,254 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
```

Comment:
the same as mnist_cnn.py?

Reply:
not exactly, mnist_cnn.py mix multiple functions in the same file "single gpu, dist_train, asychronous training", also with data augmentation.       Then doc_dist_train.py, only provide the dist train function and I removed the DIST variable for documentation use only, otherwise will be confusing if mixing all the thing into one example

Reply:
I think there is no need to create an extra doc_dist_train.py. 
In the documentation, we just show some code snippet instead of the full example.